go-controller/pkg/ovn/egressip.go

```diff
@@ -2440,6 +1612,19 @@ func ipFamilyName(isIPv6 bool) string {
 	return "ip4"
 }
 
+func (e *egressIPZoneController) getTransitSwitchIP(nodeName string) (string, error) {
+	// fetch node annotation of the egress node
+	node, err := e.watchFactory.GetNode(nodeName)
+	if err != nil {
+		return "", fmt.Errorf("failed to get node %s: %w", nodeName, err)
+	}
+	transitSwitchIP, err := util.ParseNodeTransitSwitchPortAddrs(node)
+	if err != nil {
+		return "", fmt.Errorf("unable to fetch transit switch IP for node %s: %w", nodeName, err)
+	}
+	return transitSwitchIP[0].IP.String(), nil
```

Comment:
good catch from @kyrtapz : Forgot about dualstack? :D
LOL meant to get back to this but didn't. I guess this is the side effect of working on a single PR for 4 months, you start to loose context on what all you wanted to do.